nodewatcher/plugins/sge.py

```diff
@@ -36,7 +37,53 @@ def hasJobs(hostname):
 
     return _jobs
 
-def hasPendingJobs():
+
+def get_idle_nodes():
+    command = ['/opt/sge/bin/idle-nodes']
+
+    # Command outputs the idle nodes in the following format
+    # ip-172-31-56-253
+    # ip-172-31-56-25
+    # ip-172-31-56-252
+    # ip-172-31-56-251
+
+    try:
+        process = subprocess.Popen(command, stdout=subprocess.PIPE,
+                                 env=dict(os.environ, SGE_ROOT='/opt/sge',
+                                          PATH='/opt/sge/bin:/opt/sge/bin/lx-amd64:/bin:/usr/bin'))
+    except subprocess.CalledProcessError:
+        log.error("Failed to run %s\n" % command)
+        return set()
+
+    output = process.communicate()[0]
+    idle_nodes = filter(None, output.split('\n'))
+    return idle_nodes
+
+
+def get_current_cluster_size():
+    command = "/opt/sge/bin/lx-amd64/qconf -sel"
+
+    # Command outputs the compute nodes in the following format
+    # ip-172-31-56-253.ec2.internal
+    # ip-172-31-56-252.ec2.internal
+    # ip-172-31-56-251.ec2.internal
+
+    _command = shlex.split(command)
+    try:
+        process = subprocess.Popen(_command, env=dict(os.environ, SGE_ROOT='/opt/sge',
+                                                      PATH='/opt/sge/bin:/opt/sge/bin/lx-amd64:/bin:/usr/bin'),
+                                   stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError:
```

Comment:
From the documentation it seems the CalledProcessError is raised only with `check_call` or `check_output` calls.

Then, what is the behaviour when the host is not attached to the cluster? 
Some sge commands fail with exit code 0 but with an error message, we should check the stderr for the commands.

Reply:
Same applies everywhere else the Popen method is called

Reply:
Fixed with #116